PWGHF/HFC/TableProducer/correlatorDsHadrons.cxx

```diff
@@ -213,131 +470,260 @@ struct HfCorrelatorDsHadrons {
+      int poolBin = corrBinning.getBin(std::make_tuple(c2.posZ(), c2.multFV0M()));
+      for (auto& [candidate, pAssoc] : o2::soa::combinations(o2::soa::CombinationsFullIndexPolicy(tracks1, tracks2))) {
+
+        if (yCandMax >= 0. && std::abs(yDs(candidate)) > yCandMax) {
+          continue;
+        }
+        // KKPi
+        if (candidate.isSelDsToKKPi() == selectionFlagDs) {
+          LOGF(info, "Mixed event tracks pair: (%d, %d) from events (%d, %d), track event: (%d, %d)", candidate.index(), pAssoc.index(), c1.index(), c2.index(), candidate.collision().index(), pAssoc.collision().index());
+          entryDsHadronPair(getDeltaPhi(candidate.phi(), pAssoc.phi()),
+                            candidate.eta() - pAssoc.eta(),
+                            candidate.pt(),
+                            pAssoc.pt(),
+                            poolBin);
+          entryDsHadronRecoInfo(invMassDsToKKPi(candidate), false, false);
+        }
+        // PiKK
+        if (candidate.isSelDsToPiKK() == selectionFlagDs) {
+          LOGF(info, "Mixed event tracks pair: (%d, %d) from events (%d, %d), track event: (%d, %d)", candidate.index(), pAssoc.index(), c1.index(), c2.index(), candidate.collision().index(), pAssoc.collision().index());
+          entryDsHadronPair(getDeltaPhi(candidate.phi(), pAssoc.phi()),
+                            candidate.eta() - pAssoc.eta(),
+                            candidate.pt(),
+                            pAssoc.pt(),
+                            poolBin);
+          entryDsHadronRecoInfo(invMassDsToPiKK(candidate), false, false);
+        }
+      }
+    }
+  }
+  PROCESS_SWITCH(HfCorrelatorDsHadrons, processDataME, "Process Mixed Event Data", false);
+
+  void processMcRecME(selCollisionsWithDs& collisions, candDsMcReco& candidates, candDsMcGen const& particlesMc, myTracksMc& tracks)
+  {
+    auto tracksTuple = std::make_tuple(candidates, tracks);
+    Pair<selCollisionsWithDs, candDsMcReco, myTracksMc, BinningType> pairMcRec{corrBinning, 5, -1, collisions, tracksTuple, &cache};
+
+    // bool flagDsSignal = false;
+    bool flagDsPrompt = false;
+    for (auto& [c1, tracks1, c2, tracks2] : pairMcRec) {
+      int poolBin = corrBinning.getBin(std::make_tuple(c2.posZ(), c2.multFV0M()));
+      for (auto& [candidate, pAssoc] : o2::soa::combinations(o2::soa::CombinationsFullIndexPolicy(tracks1, tracks2))) {
+        if (yCandMax >= 0. && std::abs(yDplus(candidate)) > yCandMax) {
+          continue;
+        }
+        auto prong0McPart = candidate.prong0_as<myTracksMc>().mcParticle_as<candDsMcGen>();
+        bool flagDsSignal = std::abs(candidate.flagMcMatchRec()) == 1 << DecayType::DsToKKPi;
+        // prompt
+        if (candidate.originMcRec() == RecoDecay::OriginType::Prompt) {
+          flagDsPrompt = true;
+        }
+        // non-prompt
+        if (candidate.originMcRec() == RecoDecay::OriginType::NonPrompt) {
+          flagDsPrompt = false;
+        }
```

Comment:
There is no need to have two `if` conditions here. You can decide the value the same way you do it for `flagDsSignal`.